SignalServiceKit/src/Util/OWSAnalytics.m

```diff
@@ -145,46 +162,103 @@ - (void)tryToSyncEvents
             if (!firstEventKey) {
                 return;
             }
-
+            
             firstEventDictionary = [transaction objectForKey:firstEventKey inCollection:kOWSAnalytics_EventsCollection];
             OWSAssert(firstEventDictionary);
             OWSAssert([firstEventDictionary isKindOfClass:[NSDictionary class]]);
         }];
 
-        if (!firstEventDictionary) {
-            return;
+        if (firstEventDictionary) {
+            [self sendEvent:firstEventDictionary eventKey:firstEventKey isCritical:NO];
         }
+    });
+}
 
-        DDLogDebug(@"%@ trying to deliver event: %@", self.tag, firstEventKey);
+- (void)sendEvent:(NSDictionary *)eventDictionary eventKey:(NSString *)eventKey isCritical:(BOOL)isCritical
+{
+    OWSAssert(eventDictionary);
+    OWSAssert(eventKey);
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    OWSAssert(dispatch_get_current_queue() == self.serialQueue);
+#pragma clang diagnostic pop
+
+    if (isCritical) {
```

Comment:
take it or leave it: based on our earlier discussion - I agree that "critical" is a little bit opaque. It might make sense to label it as `shouldSubmitImmediately = true` or `shouldPersist = false` or something else more to the point.